ReleaseProcess.md

```diff
@@ -37,6 +37,72 @@ This section was borrowed heavily from [Doclava release process](https://code.go
 
 See [OSSRH Publish Guide](https://central.sonatype.org/publish/publish-guide/).
 
+## Testing Maven release workflow locally
+
+The following describes how to perform the steps of the release locally to verify that they work as desired.
+
+**Warning:** Be careful with this, these steps might be outdated or incomplete. Doublecheck that you are working on a copy of your local Gson Git repository and make sure you have followed all steps. To be safe you can also temporarily turn off your internet connection to avoid accidentally pushing changes to the real remote Git or Maven repository.\
```

Comment:
This setup is a bit hacky and I am not sure if there is a proper better alternative to this, therefore I added this "Warning". But it does allow you to test the behavior of the release workflow quite accurately, which is why I added it.